jetson/nav/stateMachine.hpp

```diff
@@ -87,4 +90,12 @@ class StateMachine : public std::enable_shared_from_this<StateMachine> {
     std::shared_ptr<ObstacleAvoidanceStateMachine> mObstacleAvoidanceStateMachine;
 
     time_point mTimePoint, mPrevTimePoint;
+
+    // Recovery object
+    // TODO put in config file
+    // Test what this value should be in IRL testing
+    std::chrono::duration<int> secondsOffset = 5s;
+    double meters_threshold = 1;
+    std::chrono::duration<int> maxTurnTime = 15s;
+    std::shared_ptr<Recovery> mRecovery = std::make_shared<Recovery>(secondsOffset,meters_threshold,maxTurnTime);
```

Comment:
it doesn't seem to me like the recovery object needs multiple owners. It's owner is the state machine and its life time should be completely tied to the state machine too. Why not use a `unique_ptr`?